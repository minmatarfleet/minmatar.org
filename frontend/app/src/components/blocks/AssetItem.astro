---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import { get_item_icon } from '@helpers/eve_image_server'

import type { AssetUI } from '@dtypes/layout_components'

interface Props {
    asset: AssetUI;
}

const {
    asset
} = Astro.props

import FixedFluid from '@components/compositions/FixedFluid.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
import FlexInline from '@components/compositions/FlexInline.astro';

import VerticalCenter from '@components/blocks/VerticalCenter.astro';
---

<FixedFluid
    width='64px'
    class="[ asset-item ]"
>
    <picture>
        <img loading="lazy" src={get_item_icon(asset.id)} width="64" height="64" alt={`${asset.name} ${t('icon').toLowerCase()}`} />
    </picture>
    <FlexInline justification='space-between'>
        <VerticalCenter>
            <Flexblock gap='0' class="[ w-full ]">
                <span>
                    {asset.count} x {asset.name}
                </span>
            </Flexblock>
        </VerticalCenter>
    </FlexInline>
</FixedFluid>