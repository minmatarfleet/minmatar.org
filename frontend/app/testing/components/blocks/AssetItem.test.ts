import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AssetItem from "@components/blocks/AssetItem.astro";

test("AssetItem defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AssetItem, {
    props: {
      asset: {
        id: 91,
        name: "chicken feed",
      },
    },
  });

  expect(result).toMatchSnapshot();
});
