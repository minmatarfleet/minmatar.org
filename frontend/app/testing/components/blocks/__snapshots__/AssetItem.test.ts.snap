// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetItem defaults 1`] = `"<div class="fixed-fluid [ asset-item ]" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture> <img loading="lazy" src="https://images.evetech.net/types/91/icon?size=64" width="64" height="64" alt="chicken feed icon"> </picture> <div class="flex-inline" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: space-between;">  <div class="!flex !items-center">  <div class="flex-block [ w-full ] gap-0" data-mod style="--flex_block_gap: 0;">  <span>  x chicken feed </span>  </div>   </div>  </div>    </div>  "`;
